### DISCUSSION TITLE
Windows Package Manager 1.9.25180

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#4914] [Releases]
This release represents our first stable Windows Package Manager 1.9 build for Windows 10 (1809+), and Windows 11. 

This release includes the ability to enable experimental features unless blocked by Group Policy Objects (GPO)

---

Sixel support is here to display icons and a new progress visualization. To try out this new feature, you must be using a terminal that supports sixels, such as the [latest Windows Terminal Preview build.](https://github.com/microsoft/terminal/releases/tag/v1.22.2362.0) To enable this feature, include the following snipping in your settings file:

```json
{
  "$schema": "https://aka.ms/winget-settings.schema.json",
  "visual": {
    "enableSixels": true,
    "progressBar": "sixel"
  }
}
```
---

WinGet Configuration files with DSC Resources requiring elevation can now be handled more gracefully. The "Configure Self Elevation" experimental feature can be enabled with the following snippet in your settings file:

```json
{
  "$schema": "https://aka.ms/winget-settings.schema.json",
  "experimentalFeatures": {
    "configureSelfElevate": true
  }
}
```

Use "securityContext: elevated" under "directives:" for any resources requiring administrator. WinGet will trigger a single User Account Control (UAC) prompt when the configuration is run under user context. Resources requiring elevation are run elevated, and the rest are run in user context. The following example shows how the [.NET Tutorial - Hello World in 5 minutes](https://dotnet.microsoft.com/learn/dotnet/hello-world-tutorial/intro) will be updated to provide a better user experience.

```yaml
# yaml-language-server: $schema=https://aka.ms/configuration-dsc-schema/0.2
properties:
  resources:
    - resource: Microsoft.WinGet.DSC/WinGetPackage
      id: install-dotnet8-SDK
      directives:
        description: Install .NET 8 SDK
        securityContext: elevated
      settings:
        id: Microsoft.DotNet.SDK.8
        source: winget
    - resource: Microsoft.WinGet.DSC/WinGetPackage
      id: install-vs-code
      directives:
        description: Install Visual Studio Code
      settings:
        id: Microsoft.VisualStudioCode
        source: winget
    - resource: Microsoft.VSCode.DSC/VSCodeExtension
      id: install-csharp-dev-kit
      dependsOn:
        - install-vs-code
      directives:
        description: Install C# Dev Kit
        allowPrerelease: true
      settings:
        name: ms-dotnettools.csdevkit
        exist: true
  configurationVersion: 0.2.0
```

## What's Changed
* Increment version to 1.9 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4522
* Update localization pipeline to new auth type by @florelis in https://github.com/microsoft/winget-cli/pull/4532
* Fix loc pipeline service connection by @florelis in https://github.com/microsoft/winget-cli/pull/4533
* Enable Microsoft Entra Id auth for displaycatalog and use enterprise offline sku by @yao-msft in https://github.com/microsoft/winget-cli/pull/4544
* Add tests for configure mixed elevation by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4487
* Make v2 index EF enabled by default by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4553
* Trim Version When Parsing by @Trenly in https://github.com/microsoft/winget-cli/pull/4554
* Add Tests for Parsing Versions with Whitespace by @Trenly in https://github.com/microsoft/winget-cli/pull/4557
* Winget Repair - Eliminate installer type mapping for MSI/WIX and MSIX NonStore ,code refactoring & E2E Test Coverage by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/4534
* Update Pipeline Component Versions by @Trenly in https://github.com/microsoft/winget-cli/pull/4560
* Add support for extracting archives using tar.exe by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4541
* Make SxS stable by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4563
* Use digest verification APIs for MSIX when available by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4564
* Improve Support for Versions with Preambles by @Trenly in https://github.com/microsoft/winget-cli/pull/4558
* Launch COM server for elevated use via packaged path by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4567
* Fix export manifest with multiple locales by @yao-msft in https://github.com/microsoft/winget-cli/pull/4569
* Configuration history support by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4552
* Make Store package download stable by @yao-msft in https://github.com/microsoft/winget-cli/pull/4570
* Update expired certificate with incoming one by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4568
* Apply latest patch for localized strings by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4578
* Update winget server com security by @yao-msft in https://github.com/microsoft/winget-cli/pull/4577
* Component Governance fix: Update libyaml by @yao-msft in https://github.com/microsoft/winget-cli/pull/4583
* Update valijson to v1.0.2 by @yao-msft in https://github.com/microsoft/winget-cli/pull/4588
* Use atomic functions to ensure single initialization of tracking catalog by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4592
* Standardize line endings in expect.txt by @Trenly in https://github.com/microsoft/winget-cli/pull/4595
* Fix service connection for ESRP codesigning for utils nuget package by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4597
* Configuration apply queueing by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4590
* Ensure Agreements don't terminate in dependency flow by @Trenly in https://github.com/microsoft/winget-cli/pull/4594
* Remove libyaml Dockerfile by @florelis in https://github.com/microsoft/winget-cli/pull/4604
* Update README.md and Pull Request and Issue templates by @skanda890 in https://github.com/microsoft/winget-cli/pull/4599
* Attempt to update dependencies when not registered by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4606
* Remove experimental from index v2 by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4607
* ARM64 build and test fixes by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4608
* Use shared manifest in COM server by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4605
* Add AnyCPU to .gitignore by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4621
* Fix for attempted use of capability API before it was introduced by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4620
* Fix Component Governance by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4629
* Update documentation errors by @Trenly in https://github.com/microsoft/winget-cli/pull/4649
* Update Zlib library dependency for PureLib by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4654
* update curl vcpkg version to 8.8.0 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4655
* Pull in latest SFSClient by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4670
* Remove prerelease, update icon uri, and clean up for PowerShell modules by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4672
* Use ContinueWith rather than finally for coroutine result types by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4669
* Store configuration status and enable remote watch events by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4647
* Update README.md by @skanda890 in https://github.com/microsoft/winget-cli/pull/4687
* Update metadata collection tooling by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4688
* Skip uninstall attempt when installed package version is not present by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4685
* Upgrade to check-spelling v0.0.22 by @jsoref in https://github.com/microsoft/winget-cli/pull/3896
* Fix downloader selection by @florelis in https://github.com/microsoft/winget-cli/pull/4696
* Split testing into separate pipeline job by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4675
* Update NOTICE by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4699
* Delete CodeCoverage and WinGetUtil NuGet pipelines by @florelis in https://github.com/microsoft/winget-cli/pull/4708
* WinRT activation shim for Microsoft.Management.Deployment by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4709
* Add missing solution dependency by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4717
* Fix the actual problem with the build... by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4719
* Send unit telemetry on success by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4720
* Pull in latest SFS client by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4725
* PowerShell Help for Microsoft.WinGet.Client by @denelon in https://github.com/microsoft/winget-cli/pull/4674
* Create public sample by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4734
* Generate Microsoft.WinGet.Client help docs in cmdlet output by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4731
* Remove preference for "installed" installer type by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4740
* Update curl to 8.9.1 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4745
* Fix Reset-WinGetSource behavior by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4732
* External HR error message support by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4746
* Generate test certificates in pipeline by @florelis in https://github.com/microsoft/winget-cli/pull/4747
* [Spec] Windows Package Manager Repair functionality Design spec by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/3943
* Update build status badge by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4758
* Run test tasks if succeededOrFailed by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4759
* Binskim error EnableCriticalCompilerWarnings by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4764
* Run Unpackaged Unit Tests with winget-installed PsExec by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4760
* Spelling by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4772
* Updates to PowerShell help docs by @mdanish-kh in https://github.com/microsoft/winget-cli/pull/4781
* Repair-WinGetPackageManager improvements by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4774
* Allow for smaller configuration outputs by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4754
* Add diagnostics to dynamic factory by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4757
* PS Gallery tests are unreliable by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4791
* Add pwsh support for trust level and explicit by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4750
* Updates to the PowerShell modules by @kilasuit in https://github.com/microsoft/winget-cli/pull/4716
* Microsoft.WinGet.DSC improvements (and breaking change) by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4795
* Small build changes by @florelis in https://github.com/microsoft/winget-cli/pull/4807
* Add initial 1.9 manifest by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4811
* Fix Component Governance issues by @florelis in https://github.com/microsoft/winget-cli/pull/4814
* Fix grammatical issues in Bug Report.yml, and Update dependencies by @skanda890 in https://github.com/microsoft/winget-cli/pull/4788
* upgrading to similar-issues-ai by @denelon in https://github.com/microsoft/winget-cli/pull/4819
* Update nuget for E2E test project  by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4821
* Fix PATH issue for archive containing nested portables with binary dependencies by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4816
* Revert PowerShell modules to alpha by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4823
* Update fuzzing template area path by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4824
* Add a few more command aliases by @Trenly in https://github.com/microsoft/winget-cli/pull/4822
* Fix issue with unicode characters in symlink target path by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4834
* Pull in latest localized strings by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4840
* Sixel support by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4828
* More messages for common Store package download scenarios by @yao-msft in https://github.com/microsoft/winget-cli/pull/4841
* WinGet COM API for Repair by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/4736
* Improve configuration self elevation flow by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4844
* Pull in latest localized strings by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4855
* Update curl by @florelis in https://github.com/microsoft/winget-cli/pull/4856
* Add extended error codes to results without them by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4858
* Change enum value name by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4864
* Set WindowsSdkPackageVersion to older value by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4860
* Enable USE_PROD_CLSIDS switch in CommonCore by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4865
* Make dependency lookup case insensitive by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4866
* Documentation - added `Microsoft.Rest` as a supported source type by @bubbletroubles in https://github.com/microsoft/winget-cli/pull/4873
* Add hash mismatch telemetry details by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4857
* Update package with CVEs by @florelis in https://github.com/microsoft/winget-cli/pull/4881
* Add ADMX policies for security features in App Installer by @florelis in https://github.com/microsoft/winget-cli/pull/4726
* Consume all input during extract sequence by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4882
* Fix for Source Argument Validation in SourceWorkflow for Default Source Type by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/4891
* Add latest localized strings by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4892
* Add support for 1.9 manifest in rest source parsing and winget utils interop by @yao-msft in https://github.com/microsoft/winget-cli/pull/4906

## New Contributors
* @skanda890 made their first contribution in https://github.com/microsoft/winget-cli/pull/4599
* @kilasuit made their first contribution in https://github.com/microsoft/winget-cli/pull/4716
* @bubbletroubles made their first contribution in https://github.com/microsoft/winget-cli/pull/4873

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/v1.8.1911...v1.9.25180

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.9.25180'>Windows Package Manager 1.9.25180</a>.</em>

**Comment by psdayne** [comment]
Hey have a bunch of machines upgraded to this latest release = now running winget as system is a no go.
All winget logs have this final line...

 IsCOMCall:0; Caller: winget-cli

Even running winget via cmd line under system context returns nothing at all - cant bring up and help etc.
But works for standard user.

**Reply by denelon** [reply]
Do you get the same results with the [Microsoft.WinGet.Client](https://www.powershellgallery.com/packages/Microsoft.WinGet.Client) PowerShell module?

Note, installing packages in system context is going to need support for "System":

`Install-WinGetPackage <package id> -Scope System`

**Reply by psdayne** [reply]
On Windows 11.
The module isnt giving up anything...

![winget1](https://github.com/user-attachments/assets/3a10949b-d266-4594-b924-da3a9d153b86)

`PS C:\NotBackedUp> Get-WinGetPackage /?
Get-WinGetPackage : This cmdlet is not supported in Windows PowerShell.
At line:1 char:1
+ Get-WinGetPackage /?
+ ~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Get-WinGetPackage], WindowsPowerShellNotSupported
    + FullyQualifiedErrorId : RuntimeException,Microsoft.WinGet.Client.Commands.GetPackageCmdlet`

Even if i do get-wingetpackage -query [appname] it says the cmdlet is not supported in windows powershell.

**Reply by denelon** [reply]
What version of the module is installed?

I just installed it from the PowerShell Gallery in Windows PowerShell and ran Get-WinGetPackage (I truncated the output).

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.
                                                                                                                            Install the latest PowerShell for new features and improvements! https://aka.ms/PSWindows                                                                                                                                                               PS C:\Users\denelon> Install-Module Microsoft.WinGet.Client                                                                                                                                                                                             Untrusted repository
You are installing the modules from an untrusted repository. If you trust this repository, change its InstallationPolicy
value by running the Set-PSRepository cmdlet. Are you sure you want to install the modules from 'PSGallery'?
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "N"): Y
PS C:\Users\denelon> Get-WinGetPackage

Name                                                 Id
----                                                 --
Git                                                  Git.Git
```

**Reply by denelon** [reply]
<img width="544" alt="image" src="https://github.com/user-attachments/assets/35253df5-27a2-4eb6-a860-bac264aff4bc">

**Reply by stefnash** [reply]
I get the same issue running as SYSTEM after upgrading this morning from v1.8.1911 to v1.9.25180 - checking the version returns nothing.
![image](https://github.com/user-attachments/assets/b6dcafbb-0c36-44c3-ab53-4f175861f96c)

![image](https://github.com/user-attachments/assets/d40ec581-0d4b-4597-b5da-9a444f59c8c7)

Same **IsCOMCall:0** in the winget logs here C:\Windows\Temp\WinGet\defaultState
![image](https://github.com/user-attachments/assets/ee2ce0fc-c862-463a-8162-0ec63e0a8062)

**Comment by denelon** [comment]
* https://github.com/microsoft/winget-cli/issues/4916

**Comment by psdayne** [comment]
Your right about Vlibs being the issue.

If you install the latest 2015 VC redist then winget kicks into life from here...
`winget show Microsoft.VCRedist.2015+.x64`

[https://download.visualstudio.microsoft.com/download/pr/368cc6bf-087b-49f9-93e6-ab05b70a58e0/814E9DA5EC5E5D6A8FA701999D1FC3BADDF7F3ADC528E202590E9B1CB73E4A11/VC_redist.x64.exe](url)

Working now after deploying the VC++ Redist.

**Reply by JohnMcPMS** [reply]
We now understand this issue as being the confluence of a change to default behavior in VC builds (that started with our 1.9 release builds) combined with a Windows 11 bug whose fix is not quite released yet.  We are going to release an updated 1.9 package that uses the older build behavior to prevent this.

Installing the latest VC runtime is indeed the correct mitigation until we have released an updated package.

The scope of impact is broader than SYSTEM usage of winget.exe, which continues to be unsupported.
